### PR TITLE
nixos/tests/gitea: keep calling the file itself evaluatable

### PR DESCRIPTION
**Pull request by Ma27**
###### Description of changes
When I work on something gitea-related I want to be able to just do `nix-build nixos/tests/gitea.nix` to run the tests.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by IndeedNotJames** [comment]
I am a bit confused as to why.
Why do you need `nix-build nixos/tests/gitea.nix`?
Can't you just use `nix-build -A nixosTests.gitea` instead?

File: nixos/tests/gitea.nix [hunk 1]
```diff
@@ -1,6 +1,6 @@
 { system ? builtins.currentSystem,
   config ? {},
-  giteaPackage,
+  giteaPackage ? "gitea",
```
Comment (IndeedNotJames) [inline nixos/tests/gitea.nix:3]: I might be misunderstanding what you are trying to do here, but why not `pkgs.gitea`?

```suggestion
  giteaPackage ? pkgs.gitea,
```

Passing a string makes it much more difficult to pass an arbitrary (custom) package </3

**Review by emilylange — CHANGES_REQUESTED** [review]

File: nixos/tests/gitea.nix (hunk 1 above)
Comment (Ma27) [reply nixos/tests/gitea.nix]: Generally I don't think it's very nice to pass in a package from a different package-set, but considering that there are a bunch other tests doing the same (and >1 nixpkgs is instantiated anyways), we an save ourselves the discussion and I'll just do it :)

**Comment by IndeedNotJames** [comment]
Can you comment on my previous comment https://github.com/NixOS/nixpkgs/pull/219203#issuecomment-1452243878?
I genuinely don't understand why you need this change in the first place.

**Comment by Ma27** [comment]
First of all, when I start writing tests I don't care about the exact structure, so I'll just do `nix-build test.nix`, that's why I'm used to invoke tests like that (and I think that this is also more intuitive for newcomers, so both for usability and as a habit).
Second, having an argument that can't be autocalled (although it trivially could, because of lazy evaluation this won't even have a negative impact on the stuff defined in all-tests) isn't very nice.

**Pull request merged**